### PR TITLE
feat(app): update filter logic for yearly events

### LINKED ISSUE #49: Update yearly events logic filter
Currently by default we show all events. This should stay.

But the two checkboxes should be updated to work like this:

- by default both are unchecked and all events are shown
- if one is clicked, only events of that type are shown
- if one is clicked and another one was selected before, unselect all other checkboxes
- if one is clicked and it was selected already, unselect it and show all events again.

### PR DESCRIPTION
**Pull request by Han2-Ro**
Change the filter logic for the yearly events to toggle between Für Alle and Nur Mitglieder.
closes #49

### CODE REVIEW

File: src/components/events/YearlyEvents.svelte [hunk 1]
```diff
@@ -4,8 +4,10 @@
 
 	export let events: YVEvent[];
 
-	let for_all = true;
-	let only_members = true;
+	let filter_for_all = true;
```
Comment (MrGreenTea) [inline src/components/events/YearlyEvents.svelte:7]: I think we should talk with Emma about the wanted logic here.

I've described my understanding in #49 but that might not be what Emma had in mind. Currently there is no way to see all events at the same time though.

File: src/components/events/YearlyEvents.svelte [hunk 2]
```diff
@@ -19,17 +21,29 @@
...
+				on:click={toggleForAll}
+				checked={!filter_for_all}
+			/>
 			<label class="ml-4" for="only-members">Nur Mitglieder</label>
 		</div>
 	</div>
 	<ul class="flex flex-col lg:mt-12 lg:grid lg:grid-cols-2 lg:gap-10">
 		{#each events as event}
-			{#if (event.for_all && for_all) || (!event.for_all && only_members)}
+			{#if (event.for_all && filter_for_all) || (!event.for_all && !filter_for_all)}
```
Comment (MrGreenTea) [inline src/components/events/YearlyEvents.svelte:46]: I think it would be nicer to filter the events in the `<script />` tag and make it reactive, so something like

```ts
$: filteredEvents = events.filter(e => event.for_all === filter_for_all)
```

and then here just iterate with `each` over `filteredEvents`

That way we do less logic in the template part of the component.

File: tests/projects.spec.ts [hunk 3]
```diff
@@ -10,30 +10,55 @@ async function forceLoadImages(page: Page) {
...
-	await expect(page.getByRole('list').getByText('Nur Mitglieder')).not.toBeVisible();
+	await expectOnlyMemebersSelected(page);
 
 	await forceLoadImages(page);
 
 	await expect(page).toHaveScreenshot({ fullPage: true });
 });
 
-test('Filter for_all', async ({ page }) => {
+test('Reselect for_all', async ({ page }) => {
```
Comment (MrGreenTea) [inline tests/projects.spec.ts:54]: I think I would keep the test name as is. That way we won't have to rename the screenshots.

File: tests/projects.spec.ts-snapshots/Projects-page-screenshot-1-Mobile-Chrome-linux.png
Comment (MrGreenTea) [inline tests/projects.spec.ts-snapshots/Projects-page-screenshot-1-Mobile-Chrome-linux.png:1]: If we make the default show all events we should revert this change to the screenshots

**Review by MrGreenTea — CHANGES_REQUESTED** [review]
Let's get @thegrumpyonion input on the wanted behaviour.
Should we show all events by default and only filter when the user selects an option or should we filter by default?

File: src/components/events/YearlyEvents.svelte (hunk 1 above)
Comment (Han2-Ro) [reply src/components/events/YearlyEvents.svelte:7]: I only now realize how many different ways there are how you could do it.
Here is what Emma wrote on figma:
ich dachte das es ehr so waere das man nuer entwieder nichts oder eine sache auswaehlen kann
und das default is das keins ausgewaehlt ist

File: tests/projects.spec.ts (hunk 3 above)
Comment (Han2-Ro) [reply tests/projects.spec.ts:54]: Yeah, I just thought I had to update the screenshots anyway.

File: src/components/events/YearlyEvents.svelte (hunk 1 above)
Comment (MrGreenTea) [reply src/components/events/YearlyEvents.svelte:7]: Yeah, I talked with Emma about it and she said might have to test the different ways. She also said she likes your interpretation the most right now, so perhaps you're on the best track ;)

Comment (MrGreenTea) [reply src/components/events/YearlyEvents.svelte:7]: We'll do it like this for now, Emma said she likes it.

**Review by MrGreenTea — APPROVED** [review]
